### ISSUE TITLE
Missing warning during analysis for wrong aggregate assignment

### ISSUE THREAD

**Issue by std-max** [ghdl/ghdl#2733]
**Description**
A warning is missing while doing bad aggregate assignment.

**How to reproduce?**
```vhd :file: mwe.vhd
entity mwe is
end entity;

architecture rtl of mwe is
    signal a   : bit_vector(1 downto 0);
    signal b   : bit_vector(2 downto 0);
    signal ser : bit_vector(10 downto 0);
begin
    ser <= a & b;
end architecture;
```

```sh :image: ghdl/ghdl:bullseye-mcode
ghdl analyze --std=08 mwe.vhd
ghdl elaborate --std=08 mwe
```

There is no warning/error message while the code is wrong.
However, with the following assignment `(a, b) < ser`, I got a warning `value constraints don't match target ones [-Wruntime-error]` as intended.

**Context**
GHDL 5.0.0-dev (4.1.0.r160.gea01390c7) [Dunoon edition]
 Compiled with GNAT Version: 8.3.0
 static elaboration, mcode JIT code generator
Written by Tristan Gingold

**Comment by tgingold** [comment]
Yes, it is could be improved.  You will however get an error if you synthesize this design.
